client/rest-high-level/src/test/java/org/elasticsearch/client/license/GetBasicStatusResponseTests.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.license;
+
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractToFromXContentTestCase;
+
+import java.io.IOException;
+
+public class GetBasicStatusResponseTests
+    extends AbstractToFromXContentTestCase<GetBasicStatusResponseTests.TestResponse, GetBasicStatusResponse> {
+
+    @Override
+    protected TestResponse createTestInstance() {
+        return new TestResponse(randomBoolean());
+    }
+
+    @Override
+    protected GetBasicStatusResponse doParseInstance(XContentParser parser) {
+        return GetBasicStatusResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return true;
+    }
+
+    @Override
+    protected void assertEqualInstances(TestResponse expectedInstance, GetBasicStatusResponse newInstance) {
+        assertEquals(expectedInstance.eligibleToStartBasic, newInstance.isEligibleToStartBasic());
+    }
+
+    static class TestResponse implements ToXContentObject {
```

Comment:
@nik9000 also did soemthing like this in #33921 if you want to have a look. I think this makes more sense than putting them on the response, just because they are never used. And like we said previously, the `client ToXContent -> client FromXContent` is much less useful then `server ToXContent -> client fromXContent`, so I dont consider it a bad thing to move them here, its just making life easier than manually writing a parser.